### ISSUE TITLE
rpmlint bugs about %autochangelog

### ISSUE THREAD

**Issue by keszybz** [rpm-software-management/rpmlint#725]
```console
$ rpmlint blosc.spec
...
blosc.spec:102: W: macro-in-%changelog %autochangelog
$ rpm -q rpmlint
rpmlint-2.1.0-4.fc36.noarch
```

**Comment by marxin** [comment]
What rpmlint only does, it executes `rpm` and reports errors:

https://github.com/rpm-software-management/rpmlint/blob/5059f6202f54e5353505a9e719beace01a7ad72a/rpmlint/checks/SpecCheck.py#L625-L632

So I don't think it's issue of rpmlint.

**Comment by keszybz** [comment]
I don't have an opinion where exactly the issue should be fixed. But `%autorelease`/`%autochangelog` is slated to become the default mechanism in rpm packaging, so it'd be really awkward if `rpmlint` was warning about `%autochangelog`.

Example spec file in `fedora-rawhide-x86_64` mock:
```console
$ cat test.spec
Name: test
Version: 1
Release: %autorelease
Summary: Nothing to see
License: CC0
BuildArch: noarch
%description
%{summary}.

%prep

%build

%install

%files

%changelog
%autochangelog

$ rpmlint test.spec
test.spec:19: W: macro-in-%changelog %autochangelog

$ rpmbuild -bb test.spec
...
Wrote: /builddir/build/RPMS/test-1-1.fc36.noarch.rpm

$ rpm -qp --changelog /builddir/build/RPMS/test-1-1.fc36.noarch.rpm
* Fri Oct 08 2021 John Doe <packager@example.com> - 1-1.fc36
- local build

$ rpmlint /builddir/build/RPMS/test-1-1.fc36.noarch.rpm
...
test.noarch: W: no-url-tag
test.noarch: W: no-documentation
== 1 packages and 0 specfiles checked; 0 errors, 2 warnings, 0 badness; has taken 0.0 s ==
```
As you can see, the spec file builds fine, and once built, rpmlint is happy with the output (the warnings it emits are unrelated).

**Comment by tbaederr** [comment]
The first two you mentioned in https://github.com/rpm-software-management/rpmlint/issues/725#issuecomment-937726513 (unexpanded macro in Release), and the same one with Requires, e.g.:

```
./llvm.spec: E: specfile-error warning: line 114: Possible unexpanded macro in: Requires:	llvm-libs(x86-64) = 13.0.0-%autorelease
./llvm.spec: E: specfile-error warning: line 126: Possible unexpanded macro in: Requires:	llvm(x86-64) = 13.0.0-%autorelease
./llvm.spec: E: specfile-error warning: line 127: Possible unexpanded macro in: Requires:	llvm-libs(x86-64) = 13.0.0-%autorelease
./llvm.spec: E: specfile-error warning: line 137: Possible unexpanded macro in: Requires:	llvm-static(x86-64) = 13.0.0-%autorelease
./llvm.spec: E: specfile-error warning: line 139: Possible unexpanded macro in: Requires:	llvm-test(x86-64) = 13.0.0-%autorelease
./llvm.spec: E: specfile-error warning: line 155: Possible unexpanded macro in: Requires:	llvm = 13.0.0-%autorelease
./llvm.spec: E: specfile-error warning: line 179: Possible unexpanded macro in: Requires:	llvm(x86-64) = 13.0.0-%autorelease
./llvm.spec: E: specfile-error warning: line 180: Possible unexpanded macro in: Requires:	llvm-libs(x86-64) = 13.0.0-%autorelease
```

**Comment by marxin** [comment]
As already mentioned in https://github.com/rpm-software-management/rpmlint/issues/725#issuecomment-938502094, these should be recognized by `rpm`.

**Comment by keszybz** [comment]
Based on @tbaederr's comment, please reopen.

**Comment by marxin** [comment]
> Based on @tbaederr's comment, please reopen.

Well, as @pmatilai mentioned, the macro `%autorelease` must be somewhere defined so that `rpm` invocation can understand it when being run from `rpmlint`.
Where do you define it?

**Comment by marxin** [comment]
> Where do you define it?

https://pagure.io/fedora-infra/rpmautospec/blob/main/f/rpm/macros.d/macros.rpmautospec

**Comment by marxin** [comment]
The macro should be automatically loaded by `rpm`, so there should not be any problem from rpmlint's perspective and `specfile-error warning` should not be reported.

**Comment by marxin** [comment]
> ```
> ./fedora-obsolete-packages.spec: E: specfile-error warning: line 29: Possible unexpanded macro in: Release:    %autorelease
> ./fedora-obsolete-packages.spec: E: specfile-error warning: line 29: Possible unexpanded macro in: Release:    %autorelease
> ./fedora-obsolete-packages.spec: E: specfile-error error: %changelog entries must start with *
> ```

Btw. what rpmlint version do you use?

**Comment by marxin** [comment]
Note that since c95c6900cc1e65e841642df616903ed419b4d216, `warning`s from rpm are ignored for `specfile-error`.
